File: exercises/leetcode/1-two-sum/1-two-sum.ts
// https://leetcode.com/problems/two-sum/description/
function twoSum(nums: number[], target: number): number[] {
    const pMap: Record<number, number> = {};

    for (let i = 0; i < nums.length; i++) {
        const n = nums[i];
        const diff = target - n;

        if (diff in pMap) {
            return [pMap[diff], i];
        }

        pMap[n] = i;
    }

    return [];
}

File: exercises/leetcode/1-two-sum/1-two-sum.go
// https://leetcode.com/problems/two-sum/description/
package main

func twoSum(nums []int, target int) []int {
	pMap := map[int]int{}

	for i, n := range nums {
		diff := target - n
		if j, targetFound := pMap[diff]; targetFound {
			return []int{j, i}
		}
		pMap[n] = i
	}

	return []int{}
}
